### DISCUSSION TITLE
Welcome to HCIBench Discussions!

### DISCUSSION THREAD

**Discussion by cwei44** [vmware-labs/hci-benchmark-appliance#4] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chrgeelen** [comment]
Hello,
my name is Chris and I am using HCIBench for a couple of years now. During the last month I got sveral requests to perform SingleVM vSAN storage performance tests and the question was, if the NVMe Adapter would improve the performance of the VM. HCIBench is deploying its testing VMs with the usual Paravirtual Adapter. So, what I tried after performing the baseline test with the Paravirtual Adapter was, I shut the VM down, added the NVMe Adapter and connected the disk to the new adapter. Started the VM again and told HCIBench to re-use the existing VMs. Unfortunately, HCIBench checked the VM and deleted it and redeployed a new one. 
Is there a way to force HCIBench to re-use the existing VM? Is there a way to avoid the testing of the VM, to avoid the redeployment? Does the HCIBench testing VM Photon Image hold the right driver for the NVMe Adapter?

Since we are only purchasing Full-Flash systems fully equipped with only NVMe disks, we are would like to also only test with the NVMe adapter and driver. Your support would be very much appreciated.

Thanks for your great tool. It made my life a whole lot easier.

Best regards,
Chris

**Reply by cwei44** [reply]
Chris,

I've implemented vNVMe controller support, this is only supported on code level. 

what you can do to deploy vNVMe controller instead of pvscsi:
edit file /opt/automation/lib/deploy-vms.rb and edit
@ctrlr_type = "pvscsi" 
to 
@ctrlr_type = "nvme"

And we can generate vdbench parameter file with vNVMe controller:
run "sh /opt/automation/generate-vdb-param-file.sh" with -N it can generate vdbench parameter file in /opt/automation/vdbench-param-files/

Then you can go back to HCIBench Web UI, run testing with Vdbench. 

Please lemme know if you can get this successfully running.

Thanks
Chen
